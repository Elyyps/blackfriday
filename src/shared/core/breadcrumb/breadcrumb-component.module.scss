.breadcrumbs {
  font-size: 14px;
  font-weight: 600;

  display: flex;
  justify-content: center;
  margin-bottom: 8px;

  &__light {
    color: rgba(255, 255, 255, 0.5);
    a {
      color: rgba(255, 255, 255, 0.5);
      &:hover {
        color: $global-inverse-color;
        span {
          stroke: $global-inverse-color !important;
        }
      }
    }
    span {
      stroke: rgba(255, 255, 255, 0.5) !important;
    }

    li {
      &:nth-child(n):last-child {
        span {
          color: rgba(255, 255, 255, 0.3);
        }
      }

      &:not(:last-child) {
        &::after {
          color: rgba(255, 255, 255, 0.3);
        }
      }
    }
  }

  &__dark {
    color: $gray-light;
    a {
      color: $gray-light !important;
      &:hover {
        color: $brand-primary !important;
        span {
          stroke: $brand-primary !important;
        }
      }
    }
    span {
      stroke: $gray-light !important;
    }

    li {
      &:nth-child(n):last-child {
        span {
          color: lighten($gray-light, 10%);
        }
      }

      &:not(:last-child) {
        &::after {
          color: lighten($gray-light, 10%);
        }
      }
    }
  }

  li {
    margin: 0;
    display: inline-flex;
    align-items: center;

    &:not(:last-child) {
      &::after {
        margin-left: 8px;
        margin-right: 8px;
        display: block;
        content: "\2014";
      }
    }
  }

  ul {
    margin: 0;
  }

  a {
    text-decoration: none;
    &:hover {
      text-decoration: none;
    }
  }
}

.back-button {
  display: flex;
  align-items: center;
}

.back-button-text {
  margin-left: 8px;
}

@media (max-width: $breakpoint-xsmall-max) {
  .breadcrumb {
    justify-content: flex-start;
  }
}
