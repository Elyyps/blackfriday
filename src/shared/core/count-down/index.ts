export * from "./components/count-down.component";
