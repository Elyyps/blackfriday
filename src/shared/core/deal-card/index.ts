export * from "./deal-card.component";
