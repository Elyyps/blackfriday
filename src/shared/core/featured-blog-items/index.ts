export * from "./featured-blog-items.component";
