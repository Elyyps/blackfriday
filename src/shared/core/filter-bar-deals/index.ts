export * from "./filter-bar-deals.component";
export * from "./container";
