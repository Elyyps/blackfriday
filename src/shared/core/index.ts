export * from "./button";
export * from "./image";
export * from "./usp";
export * from "./icon";
