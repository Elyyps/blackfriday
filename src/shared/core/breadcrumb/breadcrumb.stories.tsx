import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
// import { withProvider } from "@app/modules/core/storybook-provider";
import { BreadcrumbComponent, IBreadcrumbComponentProps } from "./breadcrumb.component";

const breadcrumbItems: IBreadcrumbComponentProps = {
  links: [{ title: "Home", url: "#/" }, { title: "Blogs", url: "#/" }],
  backButton: { text: "Alle blogs", url: "/#" }
};

const styles = {
  display: "flex",
  alignItems: "center",
  background: "black",
  padding: "100px"
};

storiesOf("Breadcrumb", module)
  .addDecorator(withA11y)
  // .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <div style={styles}>
      <BreadcrumbComponent links={breadcrumbItems.links} backButton={breadcrumbItems.backButton} />
    </div>
  ));
