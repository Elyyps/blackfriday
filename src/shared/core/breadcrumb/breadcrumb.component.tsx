import * as React from "react";
import { Link } from "react-router-dom";
import { Link as LinkType } from "@app/api/core/link";
import styles from "./breadcrumb-component.module.scss";
import { IconComponent } from "../icon";
import IconBack from "@assets/icons/chevron-left.svg";

export interface IBreadcrumbComponentProps {
  backButton?: LinkType;
  links: LinkType[];
  variant?: "dark" | "light";
}

const BreadcrumbComponent = (props: IBreadcrumbComponentProps) => {
  const classNameVariant = props.variant ? `breadcrumbs__${props.variant}` : "breadcrumbs__light";
  const svgStrokeColor = props.variant && props.variant === "dark" ? "#777" : "#fff";

  return (
    <div className={`${styles["breadcrumbs"]} ${styles[classNameVariant]}`}>
      <ul className="uk-visible@s">
        {props.links.map((link, index) => (
          <li key={index}>
            {props.links.length - 1 === index ? <span>{link.title}</span> : <Link to={link.url}>{link.title}</Link>}
          </li>
        ))}
      </ul>

      {props.backButton && (
        <Link className={`uk-hidden@s ${styles["back-button"]}`} to={props.backButton.url}>
          <IconComponent icon={IconBack} size="10px" strokeColor={svgStrokeColor} />
          <span className={styles["back-button-text"]}>{props.backButton.title}</span>
        </Link>
      )}
    </div>
  );
};

export { BreadcrumbComponent };
