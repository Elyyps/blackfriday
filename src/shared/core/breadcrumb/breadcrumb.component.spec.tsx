import React from "react";
import { shallow } from "enzyme";
import { BreadcrumbComponent } from "./breadcrumb.component";

// Dummy data
import { breadcrumbItems } from "@app/api/core/breadcrumb";

describe("[Breadcrumb]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <BreadcrumbComponent links={breadcrumbItems.links} backButton={breadcrumbItems.backButton} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
