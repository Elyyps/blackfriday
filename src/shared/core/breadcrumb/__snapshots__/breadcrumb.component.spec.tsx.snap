// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Breadcrumb] should render component without crashing 1`] = `
<div
  className="breadcrumbs breadcrumbs__light"
>
  <ul
    className="uk-visible@s"
  >
    <li
      key="0"
    >
      <Link
        to="/#"
      >
        Home
      </Link>
    </li>
    <li
      key="1"
    >
      <span>
        Blog
      </span>
    </li>
  </ul>
  <Link
    className="uk-hidden@s back-button"
    to="/#"
  >
    <IconComponent
      icon="chevron-left.svg"
      size="10px"
      strokeColor="#fff"
    />
    <span
      className="back-button-text"
    >
      Terug
    </span>
  </Link>
</div>
`;
